erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -875,6 +876,12 @@ def set_tax_withholding(self):
 		# calculate totals again after applying TDS
 		self.calculate_taxes_and_totals()
 
+	def update_packing_list(self):
+		if cint(self.update_stock) == 1:
+			from erpnext.stock.doctype.packed_item.packed_item import make_packing_list
+			make_packing_list(self)
+			self.calculate_packing_list_rates()
```

Comment:
As you have added method calculate_packing_list_rates in buying_controller.py file why you're calling this method from individual file like purchase_invoice.py, purchase_order.py, purchase_receipt.py
Call the method calculate_packing_list_rates from the validate method of the buying_controller.py and put the condition of the doctype of which you want to call this method

Reply:
I did that because other docs like MaterialRequest use BuyingContoller and then I would have to check for conditions. This will add extra logic for each document extending BuyingController. If you think that adheres to erpnext guidelines better, I can do that.

Reply:
No need of extra logic you have to only add doctype condition like
```
If self.doctype in ['Purchase Order', 'Purchase Invoice', 'Purchase Receipt']:
    make_packing_list(self)
    self.calculate_packing_list_rates()
```